packages/core/src/initialize.ts

```diff
@@ -26,6 +26,8 @@ export function initializeGrafanaAgent(config: Config): Agent {
   const transports = initializeTransports(internalLogger, config);
   const api = initializeAPI(internalLogger, config, transports, metas);
 
+  api.setSession(config.session);
```

Comment:
shouldn't this be inside `initializeAPI`? Either like this or actually setting it in the initialization function of the metas API.

Reply:
My thinking is that setting session is not the concern of _initializing the api itself_ , thus does not belong there. Not setting it in metas init, as I thought it  best to always set session using this setter

Reply:
Ah, I was seeing at as "initialize with those (initial) values".